### PR TITLE
Feature/back/ar 26 animals catalog filter

### PR DESCRIPTION
**Pull request by AlexeyDerepa**

### CODE REVIEW

File: Backend/AnimalRescue/AnimalRescue.BusinessLogic/Extensions/queryExtension.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using AnimalRescue.Contracts.Query;
+
+namespace AnimalRescue.DataAccess.Contracts.Query
+{
+    public static class queryExtension
```
Comment (AlistratovAlex) [inline Backend/AnimalRescue/AnimalRescue.BusinessLogic/Extensions/queryExtension.cs]: queryExtension->QueryExtension

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/BaseCollection.cs [hunk 2]
```diff
@@ -57,41 +74,44 @@ protected Tout ConvertOneTo<Tout>(IAsyncCursor<T> item)
...
         }
+
+        public async Task<List<T>> GetAsync(DbQuery query)
+        {
+            List<T> requestedCollection = await collection
+                .Find(queryBuilder.FilterAsString(query.Filter))
+                .Sort(queryBuilder.SortAsString(query.Sort))
+                .Skip(query.Skip)
+                .Limit(query.Size)
+                .ToListAsync(); ;
```
Comment (AlistratovAlex) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/BaseCollection.cs]: Semicolumn

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Configurations/MappingProfiles/CmsConfigurationMappingProfile.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿using AnimalRescue.DataAccess.Mongodb.Models;
+using AnimalRescue.Models.DTO;
+
+using AutoMapper;
+
+namespace AnimalRescue.DataAccess.Mongodb.Configurations.MappingProfiles
+{
+    class CmsConfigurationMappingProfile : Profile
```
Comment (AlistratovAlex) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Configurations/MappingProfiles/CmsConfigurationMappingProfile.cs]: add access modificator

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Exceptions/OperatorException.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using AnimalRescue.Infrastructure.Http;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+
+namespace AnimalRescue.DataAccess.Mongodb.Exceptions
+{
+    class OperatorException : Exception, IAppException
```
Comment (AlistratovAlex) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Exceptions/OperatorException.cs]: add access modificator

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/MongoDbConfigureExtension.cs [hunk 5]
```diff
@@ -18,33 +20,37 @@
...
                 .AddSingleton<IMongoDbSettings>(p => commonSettings)
-                .AddSingleton<IBucketSettings>(p => commonSettings);
+                .AddSingleton<IBucketSettings>(p => commonSettings)
+                .AddSingleton<IMongoClient, MongoClient>(p => client)
+                .AddSingleton<IQueryFilterBuilder, QueryFilterBuilder>()
+                .AddSingleton<IQuerySortBuilder, QuerySortBuilder>()
+                .AddSingleton<IQueryBuilder<Animal>, QueryBuilder<Animal>>()
+                .AddSingleton<IQueryBuilder<Blog>, QueryBuilder<Blog>>()
+                .AddSingleton<IQueryBuilder<Configuration<CmsConfigurationNested>>, QueryBuilder<Configuration<CmsConfigurationNested>>>()
+                ;
```
Comment (AlistratovAlex) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/MongoDbConfigureExtension.cs]: move semicolon line 45

File: Backend/AnimalRescue/AnimalRescue.BusinessLogic/Extensions/queryExtension.cs (hunk 1 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.BusinessLogic/Extensions/queryExtension.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/BaseCollection.cs (hunk 2 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/BaseCollection.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Configurations/MappingProfiles/CmsConfigurationMappingProfile.cs (hunk 3 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Configurations/MappingProfiles/CmsConfigurationMappingProfile.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Exceptions/OperatorException.cs (hunk 4 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Exceptions/OperatorException.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/MongoDbConfigureExtension.cs (hunk 5 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/MongoDbConfigureExtension.cs]: done

**Pull request merged**